include/IRCMessage.hpp

```diff
@@ -17,10 +19,18 @@ class IRCMessage
 
 	void		parse();
 	void		execute();
-	
 	private:
+	
+	size_t	getArgsCount(void) const;
+	const std::string	*getArguments(void) const;
+	const std::string	&getInput(void) const;
 
-	void	_parseArguments(std::string::const_iterator &it);
+	IRCPeer	&peer;
+
+	protected:
+
+	void				_parseArguments(std::string::const_iterator &it);
+	const std::string	&_updateInputFromFields(void);
```

Comment:
will be obliterated by my PR anyway

Reply:
Your PR will never happen you ignoble piece of garbage